### PR TITLE
ENT-10009 Enhance SendTransactionFlow to allow sending a txn to multiple sessions.

### PR DESCRIPTION
**Pull request by colljos**
- Update FinalityFlow to use above.
- Remove redundant "persist" flag from `TransactionMetadata`

**Additional**
- address [PR review comment](https://github.com/corda/enterprise/pull/4805#discussion_r1222651369): start `partyInfoCache` before state machine is started.
- update outdated `api-current.txt`

### CODE REVIEW

File: core/src/main/kotlin/net/corda/core/flows/SendTransactionFlow.kt [hunk 1]
```diff
@@ -140,115 +142,126 @@ open class DataVendingFlow(val otherSideSession: FlowSession, val payload: Any,
...
-                        // Security TODO: Check for abnormally large or malformed data requests
-                        verifyDataRequest(request)
-                        request
-                    }
-                    FetchDataFlow.Request.End -> {
-                        logger.trace { "DataVendingFlow: END" }
-                        return null
+        otherSessions.forEachIndexed { idx, otherSideSession ->
+            if (payloadWithMetadata != null)
+                payload = payloadWithMetadata
```
Comment (colljos) [inline core/src/main/kotlin/net/corda/core/flows/SendTransactionFlow.kt:164]: **Note**: the implementation of the call() method has been subject to minimal change. 
The additional indentation introduced with:
`        otherSessions.forEachIndexed { idx, otherSideSession ->`
causes it to appear as vastly different.

File: core/src/main/kotlin/net/corda/core/flows/SendTransactionFlow.kt [hunk 2]
```diff
@@ -74,13 +74,14 @@ class MaybeSerializedSignedTransaction(override val id: SecureHash, val serializ
...
  *
- * @param otherSide the target party.
- * @param stx the [SignedTransaction] being sent to the [otherSideSession].
- * @property txnMetadata transaction recovery metadata (eg. used by Two Phase Finality).
+ * @param otherSessions the target parties.
+ * @param stx the [SignedTransaction] being sent to the [otherSessions].
+ * @param txnMetadata transaction recovery metadata (eg. used by Two Phase Finality).
  */
-open class SendTransactionFlow(otherSide: FlowSession, stx: SignedTransaction, txnMetadata: TransactionMetadata) : DataVendingFlow(otherSide, stx, txnMetadata) {
+open class SendTransactionFlow(otherSessions: Set<FlowSession>, stx: SignedTransaction, txnMetadata: TransactionMetadata) : DataVendingFlow(otherSessions, stx, txnMetadata) {
```
Comment (rick-r3) [inline core/src/main/kotlin/net/corda/core/flows/SendTransactionFlow.kt]: Don't think we should expect the metadata to be passed in.  I think we should be using (in `FinalityFlow`):

`SendTransactionFlow(stx: SignedTransaction, participantSessions: Set<FlowSession>, observerSessions: Set<FlowSession>, senderStatesToRecord: StatesToRecord)`

and then build the metadata automatically.  Does that work?  Ideally I'd like to remove the need to pass in the metadata.

and then the one we expect to be used outside of finality is:

`SendTransactionFlow(stx: SignedTransaction, observerSessions: Set<FlowSession>)` 

and that defaults `partitipantSessions` to the empty set and `senderStatesToRecord` to NONE.

File: node/src/main/kotlin/net/corda/node/internal/AbstractNode.kt [hunk 3]
```diff
@@ -659,6 +658,8 @@ abstract class AbstractNode<S>(val configuration: NodeConfiguration,
             tokenizableServices = null
 
             verifyCheckpointsCompatible(frozenTokenizableServices)
+            partyInfoCache.start()
```
Comment (rick-r3) [inline node/src/main/kotlin/net/corda/node/internal/AbstractNode.kt:661]: Good.

File: core/src/main/kotlin/net/corda/core/flows/SendTransactionFlow.kt (hunk 2 above)
Comment (colljos) [reply core/src/main/kotlin/net/corda/core/flows/SendTransactionFlow.kt]: Currently `SendTransactionFlow` delegates actual sending to `DataVendingFlow` which also currently takes a `TransactionMetadata`.
Where would you expect the metadata building to take place?

**Note**: `EnterpriseSendTransactionFlow` takes a slightly differently approach altogether and takes a `SignedTransactionWithTxnMetadata`as an `Any` payload passed to `EnterpriseDataVendingFlow`.
I think we should align on a single consistent approach for both.

Comment (rick-r3) [reply core/src/main/kotlin/net/corda/core/flows/SendTransactionFlow.kt]: I think we want the meta data building in `SendTransactionFlow`.

Comment (rick-r3) [reply core/src/main/kotlin/net/corda/core/flows/SendTransactionFlow.kt]: And `EnterpriseSendTransactionFlow` will just do the same.

Comment (rick-r3) [reply core/src/main/kotlin/net/corda/core/flows/SendTransactionFlow.kt]: You already do the meta data building for the simple constructor.  I'm just saying it's unlikely a user is going to know how to populate it, and we already can do so automatically.  And worse, they could populate it with wrong or malicious data.

Comment (colljos) [reply core/src/main/kotlin/net/corda/core/flows/SendTransactionFlow.kt]: OK.
`EnterpriseSendTransactionFlow` is slightly different as it takes in multiple transactions, so we feed this information in as TxnMetadata with the corresponding transaction.

**Review by rick-r3 — APPROVED** [review]

**Pull request merged**